File: Assignment18/task3.py
def factorial(n):
    """
    Calculate the factorial of a number using recursion.
    
    Args:
        n: A non-negative integer
        
    Returns:
        The factorial of n
    """
    if n < 0:
        raise ValueError("Factorial is not defined for negative numbers")
    elif n == 0 or n == 1:
        return 1
    else:
        return n * factorial(n - 1)


if __name__ == "__main__":
    try:
        num = int(input("Enter a non-negative integer: "))
        result = factorial(num)
        print(f"The factorial of {num} is {result}")
    except ValueError as e:
        print(f"Error: {e}")

File: Assignment18/task3.cpp
#include <iostream>
using namespace std;

long long factorial(int n) {
    if (n == 0 || n == 1)
        return 1;
    else
        return n * factorial(n - 1);
}

int main() {
    int num;
    cout << "Enter a non-negative integer: ";
    cin >> num;

    if (num < 0) {
        cout << "Factorial for negative numbers is not possible!";
    } else {
        long long result = factorial(num);
        cout << "Factorial of " << num << " = " << result;
    }

    return 0;
}